### ISSUE TITLE
RFE: Add a StartIntervalSec= option for services

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#7642]
### Submission type

 - Request for enhancement (RFE)

It is already possible to define the interval between process exit and process restart via the `RestartSec=` option. In some cases, however, it might be more desirable to be able to define a minimum interval between process **start** time and restart time. This could be added as an additional `StartIntervalSec=` option to the [Service] section of the unit files, or alternatively `StartIntervalMinSec=`, `RestartIntervalSec=`, ...

### Example

A unit with `Restart=always` has exited after 4m.  
With `RestartSec=5m`, the unit would restart in 5m.  
With the proposed `StartIntervalSec=5m`, it would already restart in 1m because it has already been running for 4m since its last start.

If the process runs longer than `StartIntervalSec`, the process should be restarted "immediately" (after `RestartSec`)

#### To summarize:  
`RestartSec` would be the minimum interval between process exit and process start
`StartIntervalSec` would be the minimum interval between two process starts

### See also
This is more or less what was requested in #2416, but I propose adding a new option for that to the [Service] section instead of changing the behavior of the existing `StartLimitIntervalSec` option in the [Unit] section.

**Comment by arvidjaar** [comment]
What is use case? `RestartSec` is used to rate-limit restarts; proposed settings will trigger busy loop as soon as service fails after `StartIntervalSec`.

> This is more or less what was requested in #2416

I honestly do not see how it is related.

**Comment by dtardon** [comment]
IIUC, then this is already doable using a `timer` unit with `OnUnitActiveSec=`.
